extensions/oracles/eth_deposits/deposits.go

```diff
@@ -128,8 +122,11 @@ func Start(ctx context.Context, service *common.Service, eventstore oracles.Even
 		return nil
 	})
 	if outerErr != nil {
-		service.Logger.Error("failed to listen to blocks", "error", err)
+		return fmt.Errorf("ethereum client error: %w", outerErr)
 	}
+
+	<-ctx.Done() // wait for the context to be cancelled
```

Comment:
Wait, why would we need this here? Doesn't it mess with the start of other oracles ?

Reply:
No, we run them all in separate goroutines. I guess there isn't really a need for it to wait there though, since the above function would already return in case of context cancellation